components/movementsensor/adxl345/adxl345.go

```diff
@@ -176,17 +187,28 @@ func NewAdxl345(
 	if err != nil {
 		return nil, err
 	}
-	b, err := board.FromDependencies(deps, newConf.BoardName)
+
+	bus, err := genericlinux.NewI2cBus(newConf.I2cBus)
 	if err != nil {
-		return nil, err
+		msg := fmt.Sprintf("can't find I2C bus '%q' for ADXL345 sensor", newConf.I2cBus)
+		return nil, errors.Wrap(err, msg)
 	}
-	localB, ok := b.(board.LocalBoard)
-	if !ok {
-		return nil, errors.Errorf("board %q is not local", newConf.BoardName)
-	}
-	bus, ok := localB.I2CByName(newConf.I2cBus)
-	if !ok {
-		return nil, errors.Errorf("can't find I2C bus '%q' for ADXL345 sensor", newConf.I2cBus)
+
+	// The rest of the constructor is separated out so that you can pass in a mock I2C bus during
+	// tests.
+	return makeAdxl345(ctx, deps, conf, logger, bus)
+}
+
```

Comment:
missing function description for `makeAdxl345`. It can be confusing to what the purpose of this is when we already have `NewAdxl345`